### DISCUSSION TITLE
Analog audio mic to SD card problem

### DISCUSSION THREAD

**Discussion by gvm2023** [pschatzmann/arduino-audio-tools#709] [Q&A]
Using these great audio tools , I have successfully used an I2S microphone and saved recordings as a WAV file on an SD card but when I changed over to use an analog mic, MAX9814 (I needed a long cable length for my application ) the saved WAV file results in very slow and distorted speech sounds.
It would appear that the ESP32 ADC (GPIO32) and the SD card save actions are not compatible. 
I have run out of ideas so any help would be much appreciated pointing me in the right direction. 

Basic setup 

AnalogAudioStream adc;  //was I2SStream i2s;
File audioFile;                                                   
EncodedAudioStream out(&audioFile, new WAVEncoder());             
StreamCopy copiermic(out, adc); // was StreamCopy copiermic (out, i2s);

Void loop ()
copiermic.copy();

**Comment by pschatzmann** [comment]
Analog microphones are providing a low audio quality and you need so use very short cables. 
The longer the cable the more noise!

So this is a hardware issue and has nothing to do with my software...

**Comment by gvm2023** [comment]
Thanks for the information.  I spent a few days experimenting with the I2S mic to learn that they have signal issues over 300 mm in cable length.However the issue I am having is a speed problem. I am using the MAX9824 mic with the same short length of cable as the previously working I2S mic for test purposes. The resulting WAV file sounds like the voice is in slow motion with very little noise  and while the recording length is set to 10 seconds the file save is actually taking about 13 seconds and plays for the same longer duration on my pc.The I2S mic works really well but the cable length is just too short for my application.
-------- Original message --------From: Phil Schatzmann ***@***.***> Date: 24/03/2023  17:47  (GMT+00:00) To: pschatzmann/arduino-audio-tools ***@***.***> Cc: gvm2023 ***@***.***>, Author ***@***.***> Subject: Re: [pschatzmann/arduino-audio-tools] Analog audio mic to SD card problem (Discussion #709) 
Analog microphones are providing a low audio quality and you need so use very short cables.
The longer the cable the more noise!

**Reply by gvm2023** [reply]
It is a MAX9814 microphone, 100mm cable length connected to an ESP32 WROOM 32.
It would appear that the ESP32 cannot keep up with the speed of the ADC input capture using my setup which is also affecting the  duration of the created WAV file. No issue with the I2S microphone (INMP441) file durations are exactly the same as the recording time and audio output is perfect not slow motion.

More setup details:
Sample rate = 22050 
bits per sample = 16
channels = 1
I tried reducing the sample speed with no noticeable effect.

**Comment by pschatzmann** [comment]
I'd rather think that you made a mistake with passing the sample rate or channels to the WAV. I could go up to 48000 w/o any issues.
The other thing you should look at is the spi rate of the SD drive...

**Reply by gvm2023** [reply]
The sample rate , channel and bits per sample are the same for the WAV output as for the ADC input.
I left all other settings the same as per the working I2S mic setup.
Great idea about the SPI rate , I will look at that tomorrow.  Thanks very much for your help.

**Reply by gvm2023** [reply]
I tried inceasing the frequency of the SPI but unfortunately that made no difference. the default is 4M and so I tried 8, 12 ,16 and 20.

**Reply by gvm2023** [reply]
The configuration of the adc and output to WAV is:
 auto cfgin = adc.defaultConfig(RX_MODE);
  cfgin.sample_rate = 22050;
  cfgin.bits_per_sample = 16;
  cfgin.channels = 1;
  cfgin.setInputPin1(GPIO_NUM_32);  //set GPIO 32
  adc.begin(cfgin);
  delay(100);
 
  auto cfg_out = out.defaultConfig();
  cfg_out.sample_rate = 22050;
  cfg_out.channels = 1;
  cfg_out.bits_per_sample = 16;
  out.begin(cfg_out);
  delay(100);

Do these look correct ?

**Reply by pschatzmann** [reply]
Looks fine to me, but just double check in your log if you see the expected values, when you set the log level to info

**Reply by gvm2023** [reply]
This is what I am seeing in the log:
[I] AnalogAudioBase.h : 54 - I2S_MODE_ADC_BUILT_IN
[I] AnalogAudio.h : 57 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AnalogAudioESP32.h : 48 - virtual bool audio_tools::AnalogDriverESP32::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 56 - sample_rate: 22050
[I] AudioTypes.h : 57 - channels: 1
[I] AudioTypes.h : 58 - bits_per_sample: 16
[I] AnalogAudioBase.h : 75 - input pin1: 32
[I] AnalogAudioESP32.h : 86 - RX_MODE
[I] CodecWAV.h : 455 - is_streamed! because length is 2147418112
[I] AudioCopy.h : 77 - buffer_size=1024
RECORDING
[I] CodecWAV.h : 499 - Writing Header
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
etc. to the end of recording
